### ISSUE TITLE
Transforming SVG in an <object> doesn't work on Chrome but does in Firefox

### ISSUE THREAD

**Issue by sjmurdoch** [svgdotjs/svg.js#1177]
# Bug report

## Fiddle

Please see the demonstration at https://repl.it/@sjmurdoch/svgjsdemo#index.html

## Explanation

- What is the behaviour you expect?

When requesting to transform an SVG rectangle referred to from an `<object>` tag using svg.js nothing happens in Chrome (and Safari), but does with Firefox. 

- What is happening instead?

In Chrome and Safari, the rectangle remains unchanged when modified through svg.js. Performing the same transformation through setAttribute directly works in all of Chrome, Safari, and Firefox.

- What error message are you getting?

None

**Comment by Fuzzyma** [comment]
We cant change browser behavior especially not related to the object tag.
Just use plain svg instead. You dont need `<object>`

**Comment by sjmurdoch** [comment]
Did you click on the green "Run" button at the top?

My limited understanding is that this should work even for SVG's included in an `<object>` tag and I can make it work with code that I think isn't far off from what svg.js is doing. I wasn't able to work out why there's a difference.

Here's what I see in Firefox
<img width="595" alt="svgjs-firefox" src="https://user-images.githubusercontent.com/283486/102360214-a4b0b880-3fa9-11eb-8cb7-6d966a897b65.png">

and Chrome
<img width="715" alt="svgjs-chrome" src="https://user-images.githubusercontent.com/283486/102360235-abd7c680-3fa9-11eb-838d-c041d4b8ae22.png">

**Comment by Fuzzyma** [comment]
Sorry I thought I answered already. You are right, I didnt press the run button -.-
I can see, that the transform attribute is not set in Chrome. I debugged it and found, that we do a `node instanceof window.Node` check when creating a new svgjs wrapper. This check is neccessary for convinience because its also possible to pass a plain attr object to initialize a new node. However, Chromes implementation of the object tag seems to create a different window than the "main" window and therefore their Node-Classes are not the same. The check fails, a new Node is created and this node is changed instead (you never see it because you never attach it).

I am not sure what the specs here say but it seems that one browser is right and the other one is wrong.
We might get around the issue in svg.js if we check if the plain object has no prototype but this can also go south for other reasons instead...

**Comment by sjmurdoch** [comment]
Thanks for investigating this. I'd be happy to submit a bug report to the relevant browser, but I don't know which is the correct behaviour either.

**Comment by Fuzzyma** [comment]
Haha, no worries - they will point you to the right one, if you guessed wrong :D

**Comment by wuyuanyi135** [comment]
I encountered this issue today. If `<object>` is not supported by SVG.js, could you recommend the best practice to load a SVG file served statically?

**Comment by Fuzzyma** [comment]
@wuyuanyi135 you can use ajax and pass the response to the SVG constructor and add it somewhere to your page. Alternatively you can just dump the svg into the dom

**Comment by wuyuanyi135** [comment]
@Fuzzyma thanks!

**Comment by visionm45** [comment]
So then no workaround for the object tag?  Just exploring my options and if possible that is what id like to do.

**Comment by Fuzzyma** [comment]
at least for now, no fix is planned

**Comment by visionm45** [comment]
after playing around with the nodeornew all it is is that check.  if it returned true `return node instanceof globals.window.Node ? node : create(name);` all it would take would be to change that to something else.  What could that be changed to that would make this work?

**Comment by visionm45** [comment]
okay so for all of the tests that I have done this works

`  function nodeOrNew(name, node) {
    return node && node instanceof node.ownerDocument.defaultView.Node
      ? node
      : create(name);
  } // Adopt existing svg elements`

its as if they knew we would be looking
["For example, checking if a Node is a SVGElement in a different context, you can use myNode instanceof myNode.ownerDocument.defaultView.SVGElement."](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof#instanceof_and_multiple_context_e.g._frames_or_windows)

[here is a project that I made to test everything I could think of to be sure that it works.
](https://github.com/visionm45/svgjs-object-test)


i'm sure I missed something but everything works here including manipulating an svg loaded inside of an object on chrome

**Comment by visionm45** [comment]
sorry for making so many edits to that last comment.  just to be clear I believe that I did find a solution that passes all tests except for one which is unrelated.  the code is 

`export function nodeOrNew (name, node) { 
return (node && 'ownerDocument' in node && node instanceof node.ownerDocument.defaultView.Node) ? node : create(name) 
}`

and my [branch](https://github.com/visionm45/svg.js) reflects these changes.

I hope that this can be merged at some point.

**Comment by Fuzzyma** [comment]
Was the failing test failing before you made the change? Can you also test how it affects performance by running the tests in /bench?

**Comment by visionm45** [comment]
yes I cloned master and ran the tests there and I got a the failing test mentioned above.  okay so I ran the benchmark of master and my branch and I attached the results.  I ran each test 3 times because there appeared to me to be quite a bit of variation in the results between each test.

[benchmark master.txt](https://github.com/svgdotjs/svg.js/files/7199206/benchmark.master.txt)
[benchmark branch.txt](https://github.com/svgdotjs/svg.js/files/7199207/benchmark.branch.txt)
